File: lib/aws/apigateway/clientCertificate.ts
// Copyright 2017 Pulumi, Inc. All rights reserved.

import * as cloudformation from "../cloudformation";

// The ClientCertificate resource creates a client certificate that Amazon API Gateway (API Gateway)
// uses to configure client-side SSL authentication for sending requests to the integration endpoint.
export class ClientCertificate extends cloudformation.Resource implements ClientCertificateProperties {
    public description?: string;

    constructor(name: string, args: ClientCertificateProperties) {
        super({
            name: name,
            resource: "AWS::ApiGateway::ClientCertificate",
        });
        this.description = args.description;
    }
}

export interface ClientCertificateProperties {
    // description is a description of the client certificate.
    description?: string;
}

File: lib/aws/idl/apigateway/clientCertificate.go
// Copyright 2016-2017, Pulumi Corporation.  All rights reserved.

package apigateway

import (
	"github.com/pulumi/lumi/pkg/resource/idl"
)

// The ClientCertificate resource creates a client certificate that Amazon API Gateway (API Gateway)
// uses to configure client-side SSL authentication for sending requests to the integration endpoint.
type ClientCertificate struct {
	idl.NamedResource
	// Description is a description of the client certificate.
	Description *string `lumi:"description,optional"`
}
